src/Service/SpamCheckService.php

```diff
@@ -33,36 +42,27 @@ public function __construct($apiKey, $blog)
      */
     public function isCommentAcceptable(array $data, $userIp, $userAgent)
     {
-        $comment = [];
-
-        // set some required fields
-        $comment['blog'] = $this->blog;
-
-        // TODO what are better values to use for these required fields?
-        $comment['user_ip']    = $userIp;
-        $comment['user_agent'] = $userAgent;
-
-        // now use the incoming data
-        $comment['comment_content'] = $this->getField("comment", $data);
-
-        // actually do the check
-        $ch = curl_init($this->akismetUrl . '/1.1/comment-check');
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
-        curl_setopt($ch, CURLOPT_POSTFIELDS, $comment);
-
-        $result = curl_exec($ch);
-        curl_close($ch);
-
-        // if the result is false, it wasn't spam and we can return true
-        // to indicate that the comment is acceptable
-        if ($result == "false") {
-            return true;
+        try {
+            $response = $this->httpClient->request(
+                'POST',
+                sprintf(
+                    '%s/1.1/comment-check',
+                    $this->akismetUrl
+                ),
+                [
+                    'body' => http_build_query([
+                        'blog' => $this->blog,
+                        'comment_content' => $this->getField('comment', $data),
```

Comment:
Question is: should we even bother sending a request when we already know that the comment is missing from `$data`?

Reply:
I would say no, we should skip sending the request.